aim/storage/rockscontainer.py

```diff
@@ -0,0 +1,489 @@
+import logging
+from pathlib import Path
+from filelock import FileLock
+
+import aimrocks
+
+from typing import Iterator, Optional, Tuple
+
+from aim.storage.container import Container
+from aim.storage.prefixview import PrefixView
+from aim.storage.treeview import TreeView
+
+
+logger = logging.getLogger(__name__)
+
+
+class RocksContainer(Container):
+    """
+    TODO Rocks-specific docs
+    """
+    def __init__(
+        self,
+        path: str,
+        read_only: bool = False,
+        wait_if_busy: bool = False
+    ) -> None:
+        self.path = Path(path)
+        self.read_only = read_only
+        self._db_opts = dict(
+            create_if_missing=True,
+            paranoid_checks=False,
+            keep_log_file_num=10,
+            skip_stats_update_on_db_open=True,
+            skip_checking_sst_file_sizes_on_db_open=True,
+            max_open_files=-1,
+            write_buffer_size=64 * 1024 * 1024,  # 64MB
+            max_write_buffer_number=3,
+            target_file_size_base=64 * 1024 * 1024,  # 64MB
+            max_background_compactions=4,
+            level0_file_num_compaction_trigger=8,
+            level0_slowdown_writes_trigger=17,
+            level0_stop_writes_trigger=24,
+            num_levels=4,
+            max_bytes_for_level_base=512 * 1024 * 1024,  # 512MB
+            max_bytes_for_level_multiplier=8,
+        )
+        # opts.allow_concurrent_memtable_write = False
+        # opts.memtable_factory = aimrocks.VectorMemtableFactory()
+        # opts.table_factory = aimrocks.PlainTableFactory()
+        # opts.table_factory = aimrocks.BlockBasedTableFactory(block_cache=aimrocks.LRUCache(67108864))
+        # opts.write_buffer_size = 67108864
+        # opts.arena_block_size = 67108864
+
+        self._db = None
+        self._lock = None
+        self._wait_if_busy = wait_if_busy  # TODO implement
+        self._lock_path: Optional[Path] = None
+        self._progress_path: Optional[Path] = None
+        if not self.read_only:
+            self.writable_db
+        # TODO check if Containers are reopenable
+
+    @property
```

Comment:
@mahnerak  I think there's a need to document properties as well, as it would make clear what's the exact difference between  db and writable_db

Reply:
marking this resolved as well, as @mahnerak assured that this would be done in a separate PR (for rocks specific documentation)